sdk/storage/storage-queue/karma.conf.js

```diff
@@ -78,6 +82,28 @@ module.exports = function (config) {
       properties: {} // key value pair of properties to add to the <properties> section of the report
     },
 
+    jsonToFileReporter: {
+      filter: function(obj) {
+        if (obj.writeFile) {
+          const fs = require("fs-extra");
+          // Create the directories recursively incase they don't exist
+          try {
+            // Stripping away the filename from the file path and retaining the directory structure
+            fs.ensureDirSync(obj.path.substring(0, obj.path.lastIndexOf("/") + 1));
+          } catch (err) {
+            if (err.code !== "EEXIST") throw err;
+          }
+          fs.writeFile(obj.path, JSON.stringify(obj.content, null, " "), (err) => {
+            if (err) {
+              throw err;
```

Comment:
Throw error in callback closure won’t be caught by high level method, is this by design? Or `filter` function supports a callback parameter which accepts error object?

Reply:
You're right, it won't be caught. If `obj.writeFile` is set, we're expecting the writing to succeed (that's the whole point of recording), so we are deliberately throwing if an error happens.